label_studio/blueprint.py

```diff
@@ -981,12 +981,13 @@ def main():
             ssl_context = (cert_file, key_file)
 
         # check port is busy
-        if not input_args.debug and check_port_in_use('localhost', port):
-            old_port = port
-            port = int(port) + 1
-            print('\n*** WARNING! ***\n* Port ' + str(old_port) + ' is in use.\n' +
-                  '* Trying to start at ' + str(port) +
-                  '\n****************\n')
+        if not input_args.debug:
+            while check_port_in_use('localhost', port):
+                old_port = port
+                port = int(port) + 1
+                print('\n*** WARNING! ***\n* Port ' + str(old_port) + ' is in use.\n' +
+                    '* Trying to start at ' + str(port) +
+                    '\n****************\n')
```

Comment:
Could you add here a breaking condition to avoid endless loop in the edge case?

Reply:
@leonardovida

Reply:
@makseq Sure, but if you mean when port > 65535, `socket.connect_ex()` already takes care of that raising an `OverflowError`. If you mean something else, could you please specify?

Reply:
Maybe we can limit the range to +1000 ports? I think it could take a lot of time until 65535...

Reply:
@leonardovida I think we must be hurry to be included into 0.9.1 release :smiley:

Reply:
@makseq Makes super sense limiting to +1000. I'm raising a `ConnectionError()`. Let me know if you want a more informative message or just want to `break` the loop and continue execution!

Reply:
Forgot the lack of support for fStrings. Sorry for the usage of actions